openshift_day2_configurator/configurators/ingress.py

```diff
@@ -0,0 +1,377 @@
+        )
+
+        certificate_extensions = [
+            x509.AuthorityKeyIdentifier.from_issuer_public_key(ca_key.public_key()),
+            x509.BasicConstraints(ca=False, path_length=None),
+            x509.KeyUsage(
+                digital_signature=True,
+                content_commitment=True,
+                key_encipherment=True,
+                data_encipherment=True,
+                key_agreement=False,
+                key_cert_sign=False,
+                crl_sign=False,
+                encipher_only=False,
+                decipher_only=False,
+            ),
+            x509.SubjectAlternativeName([
+                x509.DNSName(f"*.apps.{cluster_domain}"),
+                x509.DNSName(cluster_domain),
+                x509.DNSName(f"*.{cluster_domain}"),
+            ]),
+        ]
+
+        for extension in certificate_extensions:
+            certificate_builder = certificate_builder.add_extension(extension, critical=False)
+
+        ingress_certificate = certificate_builder.sign(ca_key, hashes.SHA256())
+
+        new_ingress_certificate_filename = f"*.{cluster_domain}.crt"
+        with open(new_ingress_certificate_filename, "wb") as new_ingress_certificate_file:
+            new_ingress_certificate_file.write(ingress_certificate.public_bytes(serialization.Encoding.PEM))
+
+    except Exception as ex:
+        logger.error(f"Failed to create {new_ingress_certificate_filename} Ingress certificate file: {ex}")
+        return {CREATE_NEW_INGRESS_CERTIFICATE: {"res": False, "err": str(ex)}}
+
+    return {CREATE_NEW_INGRESS_CERTIFICATE: {"res": True, "err": ""}}
+
+
+def create_ingress_certificate_configmap(
+    client: DynamicClient,
+    logger: logging.Logger,
+    cluster_domain: str,
+) -> Dict[str, Dict[str, str]]:
+    logger.debug(CREATE_INGRESS_CERTIFICATE_CONFIGMAP)
+
+    ingress_certificate = get_ingress_certificate_from_file(cluster_domain=cluster_domain)
+
+    return {
+        CREATE_INGRESS_CERTIFICATE_CONFIGMAP: create_ocp_resource(
+            ocp_resource=ConfigMap(
+                client=client,
+                name=INGRESS_CERTIFICATE_CONFIGMAP_NAME,
+                namespace="openshift-config",
+                data={
+                    "ca-bundle.crt": ingress_certificate,
+                },
+            ),
+            logger=logger,
+        )
+    }
+
+
+def update_cluster_proxy_trusted_ca(
+    client: DynamicClient,
+    logger: logging.Logger,
+) -> Dict[str, Dict[str, Union[str, bool]]]:
+    logger.debug(UPDATE_CLUSTER_PROXY_TRUSTED_CA)
+
+    cluster_proxy = Proxy(client=client, name="cluster")
+
+    if not cluster_proxy.exists:
+        cluster_proxy_not_exist_err: str = f"Cluster Proxy {cluster_proxy.name} does not exist"
+        logger.error(cluster_proxy_not_exist_err)
+        return {
+            UPDATE_CLUSTER_PROXY_TRUSTED_CA: {
+                "res": False,
+                "err": cluster_proxy_not_exist_err,
+            }
+        }
+
+    proxy_dict = {
```

Comment:
used once, no need to define and then call

Reply:
done